File: backend-java/src/main/java/com/skewwhiffy/auraltester/service/InfoService.java
package com.skewwhiffy.auraltester.service;

import com.skewwhiffy.auraltester.dto.InfoResponse;
import com.skewwhiffy.auraltester.repository.InfoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.stream.StreamSupport;

@Service
public class InfoService {
    private final InfoRepository infoRepository;

    public InfoService(@Autowired InfoRepository infoRepository) {
        this.infoRepository = infoRepository;
    }

    public InfoResponse get() {
        var allEntries = StreamSupport
                .stream(infoRepository.findAll().spliterator(), false);
        return allEntries
                .findFirst()
                .map(it -> new InfoResponse(it.getVersion()))
                .orElseThrow();
    }
}

File: backend.old/src/main/kotlin/com/skewwhiffy/auraltester/service/InfoService.kt
package com.skewwhiffy.auraltester.service

import com.skewwhiffy.auraltester.dto.InfoResponse
import com.skewwhiffy.auraltester.repository.InfoRepository
import org.springframework.stereotype.Service

@Service
class InfoService(private val infoRepository: InfoRepository) {
    fun get() = infoRepository.findAll()
        .first()
        .let { InfoResponse(it.version) }
}
